jina/peapods/pods/compoundpod.py

```diff
@@ -0,0 +1,253 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import argparse
+from argparse import Namespace
+from typing import Optional, Dict, List, Union, Set
+
+from .. import Pea
+from .. import Pod
+from .. import BasePod
+from ...enums import PollingType, SchedulerType, PodRoleType, PeaRoleType
+
+
+class CompoundPod(BasePod):
+    """A CompoundPod is a immutable set of pods, which run in parallel. They share the same input and output socket.
+    Internally, the peas of the pods can run with the process/thread backend. They can be also run in their own containers.
+    :param args: arguments parsed from the CLI
+    :param needs: pod names of preceding pods, the output of these pods are going into the input of this pod
+    """
+
+    def __init__(self, args: Union['argparse.Namespace', Dict], needs: Set[str] = None):
```

Comment:
what is the logic for these args? arr they the args that will be applied to every Pod in the Compound? Is it already a specialized argparse that takes into account?

This does not seem clear

Reply:
similar to the pod passing the arguments to the peas, the Compound pod passes the arguments to the pods

Reply:
I can make the docstring more clear

Reply:
but it will be the same argument for a single Pod reused for new Pods? will another argument Namespace will be created?

Reply:
the Namespace is deep copied

Reply:
but then, doesnt a conflict of Ports happen between replicas?

What info is in this args?